diff -Naur tensorflow-1.14.0/tensorflow/python/eager/pywrap_tensor.cc tensorflow-1.14.0-new/tensorflow/python/eager/pywrap_tensor.cc
--- tensorflow-1.14.0/tensorflow/python/eager/pywrap_tensor.cc	2019-06-18 15:48:23.000000000 -0700
+++ tensorflow-1.14.0-new/tensorflow/python/eager/pywrap_tensor.cc	2021-07-15 17:42:55.752351645 -0700
@@ -773,7 +773,7 @@
     sizeof(EagerTensor),                /* tp_basicsize */
     0,                                  /* tp_itemsize */
     (destructor)EagerTensor_dealloc,    /* tp_dealloc */
-    nullptr,                            /* tp_print */
+    0,                                  /* tp_print */
     nullptr,                            /* tp_getattr */
     nullptr,                            /* tp_setattr */
     nullptr,                            /* tp_compare */
diff -Naur tensorflow-1.14.0/tensorflow/python/eager/pywrap_tfe_src.cc tensorflow-1.14.0-new/tensorflow/python/eager/pywrap_tfe_src.cc
--- tensorflow-1.14.0/tensorflow/python/eager/pywrap_tfe_src.cc	2019-06-18 15:48:23.000000000 -0700
+++ tensorflow-1.14.0-new/tensorflow/python/eager/pywrap_tfe_src.cc	2021-07-15 17:43:18.287207437 -0700
@@ -1268,7 +1268,7 @@
     sizeof(TFE_Py_Tape),                          /* tp_basicsize */
     0,                                            /* tp_itemsize */
     &TFE_Py_Tape_Delete,                          /* tp_dealloc */
-    nullptr,                                      /* tp_print */
+    0,                                            /* tp_print */
     nullptr,                                      /* tp_getattr */
     nullptr,                                      /* tp_setattr */
     nullptr,                                      /* tp_reserved */
diff -Naur tensorflow-1.14.0/tensorflow/python/lib/core/bfloat16.cc tensorflow-1.14.0-new/tensorflow/python/lib/core/bfloat16.cc
--- tensorflow-1.14.0/tensorflow/python/lib/core/bfloat16.cc	2019-06-18 15:48:23.000000000 -0700
+++ tensorflow-1.14.0-new/tensorflow/python/lib/core/bfloat16.cc	2021-07-15 17:29:39.689333315 -0700
@@ -317,7 +317,7 @@
     sizeof(PyBfloat16),                        // tp_basicsize
     0,                                         // tp_itemsize
     nullptr,                                   // tp_dealloc
-    nullptr,                                   // tp_print
+    0,                                         // tp_print
     nullptr,                                   // tp_getattr
     nullptr,                                   // tp_setattr
     nullptr,                                   // tp_compare / tp_reserved
@@ -630,26 +630,26 @@
   const std::array<int, 3> compare_types = {
       {npy_bfloat16_, npy_bfloat16_, NPY_BOOL}};
 
-  if (!register_ufunc("equal", CompareUFunc<Bfloat16EqFunctor>,
+  if (!register_ufunc("equal", (PyUFuncGenericFunction) CompareUFunc<Bfloat16EqFunctor>,
                       compare_types)) {
     return false;
   }
-  if (!register_ufunc("not_equal", CompareUFunc<Bfloat16NeFunctor>,
+  if (!register_ufunc("not_equal", (PyUFuncGenericFunction) CompareUFunc<Bfloat16NeFunctor>,
                       compare_types)) {
     return false;
   }
-  if (!register_ufunc("less", CompareUFunc<Bfloat16LtFunctor>, compare_types)) {
+  if (!register_ufunc("less", (PyUFuncGenericFunction) CompareUFunc<Bfloat16LtFunctor>, compare_types)) {
     return false;
   }
-  if (!register_ufunc("greater", CompareUFunc<Bfloat16GtFunctor>,
+  if (!register_ufunc("greater", (PyUFuncGenericFunction) CompareUFunc<Bfloat16GtFunctor>,
                       compare_types)) {
     return false;
   }
-  if (!register_ufunc("less_equal", CompareUFunc<Bfloat16LeFunctor>,
+  if (!register_ufunc("less_equal", (PyUFuncGenericFunction) CompareUFunc<Bfloat16LeFunctor>,
                       compare_types)) {
     return false;
   }
-  if (!register_ufunc("greater_equal", CompareUFunc<Bfloat16GeFunctor>,
+  if (!register_ufunc("greater_equal", (PyUFuncGenericFunction) CompareUFunc<Bfloat16GeFunctor>,
                       compare_types)) {
     return false;
   }
diff -Naur tensorflow-1.14.0/tensorflow/python/lib/core/ndarray_tensor_bridge.cc tensorflow-1.14.0-new/tensorflow/python/lib/core/ndarray_tensor_bridge.cc
--- tensorflow-1.14.0/tensorflow/python/lib/core/ndarray_tensor_bridge.cc	2019-06-18 15:48:23.000000000 -0700
+++ tensorflow-1.14.0-new/tensorflow/python/lib/core/ndarray_tensor_bridge.cc	2021-07-15 17:42:13.996912997 -0700
@@ -86,7 +86,7 @@
     0,                                /* tp_itemsize */
     /* methods */
     TensorReleaser_dealloc,      /* tp_dealloc */
-    nullptr,                     /* tp_print */
+    0,                           /* tp_print */
     nullptr,                     /* tp_getattr */
     nullptr,                     /* tp_setattr */
     nullptr,                     /* tp_compare */
